lib/rails_state_machine/state_machine.rb

```diff
@@ -138,5 +145,12 @@ def define_model_methods
         end
       end
     end
+
+    def prefix
+      if @prefix_for_state_name.present?
+        prefix = @prefix_for_state_name.upcase.gsub(' ', '_')
+        "#{prefix}_"
+      end
+    end
```

Comment:
This works only for constant definition, not for method names.

It's also fairly unexpected that reading `prefix` from the state machine instance returns a very different value than the one it was initialized with.

Maybe this should be `prefix_for_constant_definition`.

I also suggest "caching" the result in an instance variable to avoid recalculation each time it's called.

And I think we really don't need to support spaces in prefix strings. State names also can not include spaces , so it's somewhat inconsistent to existing behavior. I assume its not a feature most would expect to exist in the first place.